# =====================
# axis_1: Turnigy KV350
# =====================

axis_1:

  # ----------------
  # basic properties
  # ----------------

  reduction: 50.0
  direction: 1 # 0: +(counter-clockwise) 1: -(clockwise) 

  # --------
  # start-up
  # --------

  startup_encoder_index_search: True

  # -------------
  # motor related
  # -------------

  motor_config:
    pre_calibrated: True
    pole_pairs: 7
    # current limit
    current_lim: 10.0 # (A)
    calibration_current: 10.0 # (A)
    current_lim_tolerance: 1.25

  # ---------------
  # encoder related
  # ---------------

  encoder_config:
    pre_calibrated: True
    cpr: 4000
    use_index: True

  # ----------
  # controller
  # ----------

  controller_config:
    vel_limit: 5000.0 # (encoder counts / sec)
    vel_limit_tolerance: 1.2
    pos_gain: 200.0
    vel_gain: 0.0006
    vel_integrator_gain: 0.0
    control_mode: 3 # position control (default)
